cmd/nfd-master/main.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/docopt/docopt-go"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"sigs.k8s.io/node-feature-discovery/pkg/apihelper"
+	pb "sigs.k8s.io/node-feature-discovery/pkg/labeler"
+	"sigs.k8s.io/node-feature-discovery/pkg/version"
+)
+
+const (
+	// ProgramName is the canonical name of this program
+	ProgramName = "nfd-master"
+
+	// NodeNameEnv is the environment variable that contains this node's name.
+	NodeNameEnv = "NODE_NAME"
+
+	// Namespace for feature labels
+	labelNs = "feature.node.kubernetes.io/"
+
+	// Namespace for all NFD-related annotations
+	annotationNs = "nfd.node.kubernetes.io/"
+)
+
+// package loggers
+var (
+	stdoutLogger = log.New(os.Stdout, "", log.LstdFlags)
+	stderrLogger = log.New(os.Stderr, "", log.LstdFlags)
+)
+
+// Labels are a Kubernetes representation of discovered features.
+type Labels map[string]string
+
+// Annotations are used for NFD-related node metadata
+type Annotations map[string]string
+
+// Command line arguments
+type Args struct {
+	labelWhiteList *regexp.Regexp
+	noPublish      bool
+	port           int
+}
+
+func main() {
+	// Assert that the version is known
+	if version.Get() == "undefined" {
+		stderrLogger.Fatalf("version not set! Set -ldflags \"-X sigs.k8s.io/node-feature-discovery/pkg/version.version=`git describe --tags --dirty --always`\" during build or run.")
+	}
+	stdoutLogger.Printf("Node Feature Discovery Master %s", version.Get())
+
+	// Parse command-line arguments.
+	args := argsParse(nil)
+
+	helper := apihelper.APIHelpers(apihelper.K8sHelpers{AnnotationNs: annotationNs,
+		LabelNs: labelNs})
+
+	if !args.noPublish {
+		err := updateMasterNode(helper)
+		if err != nil {
+			stderrLogger.Fatalf("failed to update master node: %v", err)
+		}
+	}
+
+	// Create server listening for TCP connections
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", args.port))
+	if err != nil {
+		stderrLogger.Fatalf("failed to listen: %v", err)
+	}
+	grpcServer := grpc.NewServer()
+	pb.RegisterLabelerServer(grpcServer, &labelerServer{args: args, apiHelper: helper})
+	grpcServer.Serve(lis)
+}
```

Comment:
Can we add here more info like 
```suggestion
	stdoutLogger.Printf("Node Feature Discover Master gRPC Server serving on port: %d", args,port)
}
```
In case someone uses a different port per commandline but forgets to update the Service.

Reply:
Makes sense. Added in the latest version of this PR.

Other suggestions around logging/verbosity are also welcome :smile: